common/app/services/video-channel.ts

```diff
@@ -230,11 +243,37 @@ export default class VideoChannel {
                         callback(toggleSubtitleTrackInListMessage.track);
                     }
                     break;
+                case 'subtitlesUpdated':
+                    const subtitlesUpdatedMessage = event.data as SubtitlesUpdatedFromVideoMessage;
+
+                    for (const callback of that.subtitlesUpdatedCallbacks) {
+                        callback(subtitlesUpdatedMessage.updatedSubtitles);
+                    }
+                    break;
                 case 'loadFiles':
                     for (const callback of that.loadFilesCallbacks) {
                         callback();
                     }
                     break;
+                case 'requestSubtitles':
```

Comment:
Same here, I don't think this protocol is necessary. `SubtitlePlayer` will automatically push the colors to `VideoPlayer` as long as `SubtitleModel` carries the colors.